src/main/java/com/amazonaws/encryptionsdk/kms/AwsKmsDataKeyEncryptionDao.java

```diff
@@ -126,11 +133,13 @@ public DecryptDataKeyResult decryptDataKey(EncryptedDataKey encryptedDataKey, Cr
         final com.amazonaws.services.kms.model.DecryptResult kmsResult;
 
         try {
-            kmsResult = getClientByKeyId(AwsKmsCmkId.fromString(providerInformation), clientSupplier)
-                    .decrypt(updateUserAgent(new DecryptRequest()
-                            .withCiphertextBlob(ByteBuffer.wrap(encryptedDataKey.getEncryptedDataKey()))
-                            .withEncryptionContext(encryptionContext)
-                            .withGrantTokens(grantTokens)));
+            kmsResult = client.decrypt(updateUserAgent(
+                new DecryptRequest()
+                    .withCiphertextBlob(ByteBuffer.wrap(encryptedDataKey.getEncryptedDataKey()))
+                    // provide the encrypted data key’s provider info as part of the AWS KMS Decrypt API call
+                    .withKeyId(providerInformation)
+                    .withEncryptionContext(encryptionContext)
+                    .withGrantTokens(grantTokens)));
```

Comment:
To be included here or in #223 we need to check that the KMS response has the correct keyId

Reply:
Good catch. Fixing.